File: backend/Application/Configuration/GameSetupDefaults.cs
namespace backend.Application.Configuration;

/// <summary>Default board shape and field values for a newly created draft game.</summary>
public static class GameSetupDefaults
{
    public const int Rows = 5;
    public const int Cols = 5;

    private static readonly int[] RowCosts = [100, 125, 150, 175, 200];

    public static IReadOnlyList<int> DefaultRowCosts => RowCosts;

    public static string[] BuildRowLabels()
    {
        return Enumerable.Range(0, Rows).Select(row => GetRowCost(row).ToString()).ToArray();
    }

    public static string[] BuildColumnLabels()
    {
        return Enumerable.Range(0, Cols).Select(GetColumnLabel).ToArray();
    }

    public static string GetColumnLabel(int columnIndex) => (columnIndex + 1).ToString();

    public static int GetRowCost(int rowIndex)
    {
        if (rowIndex < RowCosts.Length)
        {
            return RowCosts[rowIndex];
        }

        var lastCost = RowCosts[^1];
        var extraRows = rowIndex - RowCosts.Length + 1;
        return lastCost + extraRows * 25;
    }
}

File: frontend/src/features/game-setup/model/game-setup-defaults.ts
const GAME_SETUP_DEFAULT_ROW_COSTS = [100, 125, 150, 175, 200] as const

export function defaultGameSetupRowCost(rowIndex: number): number {
  if (rowIndex < GAME_SETUP_DEFAULT_ROW_COSTS.length) {
    return GAME_SETUP_DEFAULT_ROW_COSTS[rowIndex]
  }

  const lastCost = GAME_SETUP_DEFAULT_ROW_COSTS[GAME_SETUP_DEFAULT_ROW_COSTS.length - 1]
  const extraRows = rowIndex - GAME_SETUP_DEFAULT_ROW_COSTS.length + 1
  return lastCost + extraRows * 25
}

export function defaultGameSetupRowLabel(rowIndex: number): string {
  return `${defaultGameSetupRowCost(rowIndex)}`
}

export function defaultGameSetupColumnLabel(columnIndex: number): string {
  return `${columnIndex + 1}`
}
